pkg/http/clients/base_api_client.go

```diff
@@ -178,10 +178,14 @@ func (c baseAPIClient) DoRequestWithResponse(ctx context.Context, method, path s
 
 	// let the backoff plan handle the max case and context cancel
 	// WithMaxRetries is not thread-safe, so we initialize it here
+	attempts := c.maxAttempts
+	if c.maxAttempts > 0 {
```

Comment:
should we protect against negative values too?

Reply:
it is a unit64, can't be negative